flatsurf/geometry/hyperbolic.py

```diff
@@ -11098,8 +11098,8 @@ def _normalize_euclidean_boundary(self):
         We use a relatively standard randomized optimization approach to find a
         point in the intersection: we randomly shuffle the half spaces and then
         optimize a segment on some boundary of the half spaces. The
-        randomization makes this a linear time algorithm, see e.g.,
-        https://www2.cs.arizona.edu/classes/cs437/spring21/Lecture4.pdf.
+        randomization makes this a linear time algorithm, see e.g., De Berg,
+        Mark, et al. Computational geometry, Section 4.4.
```

Comment:
Maybe add comma after "et al.", and maybe keep the url too, unless it is present somewhere else.

Reply:
> keep the url too,

What do you mean? The PDF is gone and it's also not on the archive.org.

Reply:
Only ending the url with ".pdf." gets me "Page not found"; stopping at ".pdf" with no final dot works for me.

Reply:
Oh. It was just temporarily down. Now it works again with the .pdf even.